ysh/func_proc.py

```diff
@@ -369,15 +351,31 @@ def _BindTyped(
         if rest:
             lval = LeftName(rest.name, rest.blame_tok)
 
-            rest_val = value.List(pos_args[num_params:])
-            mem.SetLocalName(lval, rest_val)
+            rest_val = pos_args[num_params:-1] if block_param else pos_args[num_params:]
+            num_params += len(rest_val)
+            mem.SetLocalName(lval, value.List(rest_val))
         else:
-            if num_args > num_params:
+            if num_args > num_params + (1 if block_param else 0):
```

Comment:
Likewise can we remove `if block_param` by handling it first?

I might have missed something